docxcompose/composer.py

```diff
@@ -624,3 +647,42 @@ def fix_section_types(self, doc):
         first_new_section_idx = len(self.doc.sections) - len(doc.sections)
         self.doc.sections[first_new_section_idx].start_type = self.doc.sections[-1].start_type
         self.doc.sections[-1].start_type = doc.sections[-1].start_type
+
+    def fix_header_and_footers(self, doc):
+        """
+        The master document usually only has one section, hence its section
+        properties are defined directly in the body of the document and apply
+        to the last section of the document. For all other sections but the
+        last one, section properties are defined in the last paragraph of
+        the section.
+        Headers and footers are inherited from the previous section properties
+        if they are not defined in a given section. If not defined in the first
+        section, then blank headers and footers will be used., so we need to
+        make sure to add the definition from the main template in the first
+        section of the document if there are more than one sections.
+        """
+
+        if self.first_section_properties_added:
+            return
+
+        if len(self.doc.sections) == 1 or len(doc.sections) == 1:
+            return
+
+        first_new_section_idx = len(self.doc.sections) - len(doc.sections)
+
+        for footer_name in ('footer', 'even_page_footer', 'first_page_footer'):
+            footer_main = getattr(self.doc.sections[-1], footer_name)
+            if not footer_main._has_definition:
+                continue
+            footer_sec = getattr(self.doc.sections[first_new_section_idx], footer_name)
+            rid = footer_main._sectPr.get_footerReference(footer_main._hdrftr_index).rId
+            footer_sec._sectPr.add_footerReference(footer_main._hdrftr_index, rid)
+
+        for header_name in ('header', 'even_page_header', 'first_page_header'):
+            header_main = getattr(self.doc.sections[-1], header_name)
+            if not header_main._has_definition:
```

Comment:
do we always have a `header_main`? Or could it also be `None`

Reply:
I'm not 100% sure, but I think so. headers and footers are inherited from the previous section and in the first section, if it is not set, a blank header / footer is used. I think I'll make it safe just in case.